_posts/2020-10-01-living-the-devops-dream.md

```diff
@@ -0,0 +1,268 @@
+> * We went from concept to launch in 4.5 months / 167 days
+> * We completed 332 issues and 85% first pass complete (15% rework)
+> * We had very low WIP and good average cycle time of 2.5 days
+> * By DORA’s own metrics we achieved **Elite** status:
+>     * Deployment Frequency: We averaged 2.5 deployments per day, with a maximum of 9
+>     * Lead Time to production: averaged 52 minutes
+>     * Change Failure Rate: < 5%
+>     * Time To Restore: N/A<sup>*</sup>
+
+<sup>_* We only had one issue where we chose to invoke our incident response process.  The issue was identified out of hours and did not affect user function so investigation/resolution was deferred to the next business day.  Therefore we chose to include it in our change failure rate, but excluded from Time to Restore.  I shall cover the incident in greater detail in a future post._</sup>
+
+But how did we achieve such impressive metrics?  
+
+By living the DevOps Dream of course!
+
+## Data Driven
+
+At Mechanical Rock we believe that the key to successful product delivery is the capability to make data-driven decisions based on experiential learning.  We put the right metrics in place and have the discipline to respond to what we learn.
+
+At the highest level, our product goals and accompanying hypotheses provide the barometer for success.  But such trailing indicators are insufficient for agile course correction during delivery.
+
+Rather, we focus on leading indicators.  Core to our data driven approach are the [Software Delivery and Organisational Performance (SDO)](/2020/08/19/ignore-these-five-signals.html) metrics:
+* Lead Time
+* Deployment Frequency
+* Time to Resolution
+* Change Failure Rate
+
+We capture the Four Key metrics directly from our Delivery Pipeline and operational environment.
+
+We reviewed our progress during regular retrospectives in order to maintain [Elite status throughout](https://cloud.google.com/blog/products/devops-sre/the-2019-accelerate-state-of-devops-elite-performance-productivity-and-scaling).
+
+[![Goodharts Law - When a measure becomes a target, it ceases to be a good measure](/img/devops-dream/goodharts-law.webp)](https://sketchplanations.com/goodharts-law)
+
+A benefit of the SDO metrics is that they are difficult to game.  Game your lead time and you will impact your change failure rate.  
+
+We prefer to use metrics to understand the health of our delivery process.  To gain a deeper understanding, we also monitor other detailed measures to give us the story behind the headlines.  These include:
+* [Code Coverage](/2018/01/08/high-code-coverage-is-not-yak-shaving.html)
+* [Cycle Time](https://support.atlassian.com/jira-software-cloud/docs/view-and-understand-the-control-chart/)
+* [Cumulative Flow](https://support.atlassian.com/jira-software-cloud/docs/view-and-understand-the-cumulative-flow-diagram/)
+
+Using these measures, we eliminate waste within our delivery process, amplify our feedback loops, and continually learn.
```

Comment:
amplify our feedback loops? That doesn't sound right...(??)

Reply:
This is a direct reference to the three ways.  

I don't see the issue myself - will leave as is.